addons/mass_mailing/tests/test_mailing_internals.py

```diff
@@ -193,6 +193,31 @@ def test_mailing_computed_fields_form(self):
         )
         self.assertEqual(mailing_form.mailing_model_real, 'res.partner')
 
+    @mute_logger('odoo.sql_db')
+    @users('user_marketing')
+    def test_mailing_trace_values(self):
+        recipient = self.env['res.partner'].create({
+            'name': 'Mass Mail Partner',
+            'email': 'Customer <test.customer@example.com>',
+        })
+
+        with self.assertRaises(IntegrityError):
+            self.env['mailing.trace'].create({
+                'model': recipient._name,
```

Comment:
maybe it would be better to add `res_id: 0` ? (here it failed because of the `required=True`, not because of your new constraint)

Reply:
Added all use cases to be sure. More is less. Or the inverse. Bref, this should not take too much computation time on runbot.

Reply:
Added creating with a res_id and not a res_model. Just to be sure someone does not stupid things with models.